### ISSUE TITLE
Make changes to allow some interaction with the IoBroker platform.

### ISSUE THREAD

**Issue by pjaos** [pjaos/ct6_meter_os#5]
In order to make this possible the CT6 unit needs to send JSON data to an MQTT server. The following is a list of the features required in order to make this possible. 

- Set WiFi SSID and password
- Set CT1,2,3,4,5 & 6 port names
- Set the IP address/port to send JSON messages containing data from each CT port. I believe this is all the MQTT server would require. Once received by the server you'd need to sort out how this was stored and displayed. I assume this is all part of the IoBroker package.

**Comment by pjaos** [comment]
I decided that it would be best to get the apps to work on a Windows machine as well as Linux. I'm almost there with these changes. Once this is complete all the code will be available on Windows making it simpler to build the required features to satisfy the requirements of this issue.

**Comment by CNC-Maniac** [comment]
Thanks for your work!!!

**Comment by pjaos** [comment]
No problem, hopefully people will find it useful. I'll push the changes to the master branch when it's ready. I plan to store the windows installer in the repo to make it easier to install.

Paul

⁣Get BlueMail for Android ​

**Comment by pjaos** [comment]
I've completed the implementation of the changes to support the requirements of this issue. Testing and documentation is now underway.  Any help with the testing gratefully received although I probably need to get some of the documentation updated before I can pass this over.

It's taken longer than expected but I've added a Web GUI that provides the following features.

- Set the WiFi credentials of a CT6 unit via the serial port on the CT6 unit.
- Upgrade the CT6 firmware over the WiFi network.
- Set the CT6 Unit name.
- Set The CT6 port names.
- Set the address, port and Topic so that the CT6 unit pushes JSON data to an MQTT server.
- - Recover a CT6 unit that will not boot by wiping the flash memory, reloading MicroPython and the CT6 firmware as well as reloading the factory config file onto the CT6 unit.

This Web GUI runs on Linux and Windows machines.
